### PR TITLE
EZP-23831: As a UI dev, I need format of RichText field over REST to be semantically enhanced xhtml5

### PR DESCRIPTION
**Pull request by pspanja**
This PR resolves https://jira.ez.no/browse/EZP-23831

This replaces usage of custom elements in `xhtml5edit` format with standard XHTML5 elements.
Replaced elements are only those relating to embeds:

* ezlink => link
* ezconfig => span
* ezvalue => span
* ezembed => div
* ezembedinline => span

Template elements are also affected but skipped here as they are not in the current scope from the UI side and there seems to be some questions to answer there first.

Using same elements in different context is done through `data-ezelement` attribute. A format like that can't be properly described using XSD, which will become a problem when template elements are handled in the same way. This could be described using RelaxNG and/or Schematron.

### CODE REVIEW

**Comment by pspanja** [comment]
Review ping @dpobel @andrerom @bdunogier @lolautruche

POV on schema appreciated, do we need it here in the first place?

File: eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl [hunk 1]
```diff
@@ -436,39 +436,59 @@
...
-      </xsl:if>
-      <xsl:if test="@ezxhtml:align">
-        <xsl:attribute name="data-ezalign">
-          <xsl:value-of select="@ezxhtml:align"/>
-        </xsl:attribute>
-      </xsl:if>
+  <xsl:template match="docbook:ezembed">
+    <xsl:element name="div" namespace="{$outputNamespace}">
+      <xsl:attribute name="data-ezelement">
+        <xsl:value-of select="'ezembed'"/>
```
Comment (dpobel) [inline eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl]: nitpick: can be simplified to 
```xml
<xsl:attribute name="data-ezelement">ezembed</xsl:attribute>
```

**Comment by lolautruche** [comment]
Looks good AFAICT

File: eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-      <ezvalue key="size">medium</ezvalue>
-      <ezvalue key="offset">10</ezvalue>
-      <ezvalue key="limit">5</ezvalue>
-    </ezconfig>
-  </ezembed>
-  <ezembed href="ezlocation://601" id="embed-id-2" data-ezview="line" class="embedClass2" data-ezalign="right">
-    <ezlink href="ezlocation://202" class="linkClass2"/>
-  </ezembed>
+  <div data-ezelement="ezembed" href="ezcontent://106" id="embed-id-1" data-ezview="embed" class="embed-class" data-ezalign="left">
+    <a data-ezelement="ezlink" href="ezurl://95#fragment1" target="_blank" id="link-id-1" title="Link title" class="link-class"/>
```
Comment (dpobel) [inline eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml]: it's a weird markup, to improve it a bit, I would rather use a `link` tag here. According to http://www.w3.org/html/wg/drafts/html/master/semantics.html#the-link-element, it should have an `itemprop` attribute, I suggest to set its value to `url` and that also means the `div` has to have an `itemscope` attribute as well.

**Comment by andrerom** [comment]
Looks, good so far, I'm a bit concerned by usage of same tag for different things, but I'm also aware we should stay as close to the format supported by editor as possible to avoid tons of issues over the next couple of years *(see ezoe)*.

File: eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml (hunk 2 above)
Comment (pspanja) [inline eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml]: Isn't that element supposed to occur only in metadata content (head)?

Comment (andrerom) [inline eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml]: see: http://www.w3.org/html/wg/drafts/html/master/dom.html#phrasing-content-2

File: eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl (hunk 1 above)
Comment (pspanja) [inline eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl]: Simplified in https://github.com/ezsystems/ezpublish-kernel/commit/0f2c9f69c90e5fc44feefca3196b03fd3a781115

File: eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml (hunk 2 above)
Comment (pspanja) [inline eZ/Publish/Core/FieldType/Tests/RichText/Converter/Xslt/_fixtures/xhtml5/edit/027-linkedEmbed.xml]: Got it, updated in https://github.com/ezsystems/ezpublish-kernel/commit/4eba9311038733d335fa77cb93177a43a5e3b23f.

**Comment by pspanja** [comment]
Removed schema for `http://ez.no/namespaces/ezpublish5/xhtml5/edit` in https://github.com/ezsystems/ezpublish-kernel/commit/1da8dd254a1926591c0519dfd4dff2c7dffe3178.
I assumed everyone is ok with that.

**Comment by pspanja** [comment]
All remarks taken care of BTW :)

`git review --continue`

**Comment by glye** [comment]
+1 fwiw, simplification sounds good.

**Comment by andrerom** [comment]
+1, *side: should we consider renaming this format to `xhtml5-semantic` or something?*

**Comment by pspanja** [comment]
@andrerom I was thinking something in the `xhtml5-edit-alloy` line :)

**Comment by andrerom** [comment]
> I was thinking something in the xhtml5-edit-alloy line :)

ok, lets leave the name discussion for later, get your point :)

**Pull request merged**